c++/nda/lapack.hpp

```diff
@@ -54,8 +54,8 @@ namespace nda::lapack {
    * @param ipiv  Gauss Pivot, cf lapack doc
    *
    */
-  template <typename M>
-  [[nodiscard]] int getrf(M &&m, array<int, 1> &ipiv) {
+  template <typename M, typename CP>
+  [[nodiscard]] int getrf(M &&m, basic_array<int, 1, C_layout, 'A', CP> &ipiv) REQUIRES(is_ndarray_v<std::decay_t<M>> and (get_rank<std::decay_t<M>> == 2)) {
```

Comment:
We can static_assert instead of REQUIRES here, there is no other option, right ?

Reply:
We could also generalize that to array of rank 2?
static_assert for the rank check is fine with me. In that case I prefer the explicit compiler error in fact.

Reply:
static_assert for both . why not, there is no other overload ...
Also, only the error in C++20 really matters... C++17 is backward mode.

Reply:
Ok